src/components/NavigationBar/UserInfo/client.tsx

```diff
@@ -24,8 +25,29 @@ export const UserInfoClient = ({
   userInfoNode,
 }: UserInfoClientProps) => {
   const pathname = usePathname();
+  const [shouldDisplayError, setShouldDisplayError] = useState(false);
   const [logoutData, logout] = useActionState(actionLogout, { ok: true });
   const roleColor = getCodefunRoleTextClass(role);
+
+  useEffect(() => {
+    if (!logoutData.ok) {
+      setShouldDisplayError(true);
+    } else {
+      setShouldDisplayError(false);
+    }
+  }, [logoutData.ok, setShouldDisplayError, logoutData]);
+
+  useEffect(() => {
+    if (shouldDisplayError) {
+      const timeout = setTimeout(() => {
+        setShouldDisplayError(false);
+      }, 5000);
+      return () => {
+        clearTimeout(timeout);
+      };
+    }
+  }, [shouldDisplayError, setShouldDisplayError]);
+
```

Comment:
Can these two effects... be combined? This looks like an unnecessary extra render to me.

Reply:
Will try to next time 😄